### DISCUSSION TITLE
Comparison with QtCharts and QtDataVisualization?

### DISCUSSION THREAD

**Discussion by apwebber** [pyqtgraph/pyqtgraph#2183] [General]
Hi, I've been using pyqtgraph quite a bit recently and I like it a lot. I noticed that on the website pyqtgraph is compared with other popular python graphing libraries like matplotlib. But what about the native Qt graphing modules QtCharts and QtDataVisualisation? I haven't used either but PyQt5 and 6 provide bindings for them and they look very powerful. How does pyqtgraph compare?

**Comment by j9ac9k** [comment]
Hi @apwebber,

I only briefly used QtCharts and haven't used QtDataVisualization at all, so the comparison will be quick and just a gloss over. 

QtCharts is generally intended for charts requiring little to no interactivity.  It does simple static plots really well with a relatively simple API.  A lot more plot types are actually offered than pyqtgraph offers.  Doing mouse events (zoom, pan, hover) is a lot trickier.  QtCharts uses the GraphicsView framework which is what pyqtgraph uses as well.

I haven't used QtDataVisualization, but at a glance it appears intended for 3D plots and it appears to only be used with QtQuick; which likely makes it incompatible with QtWidgets, which is where pyqtgraph operates in.  It looks to be something introduced in Qt5, and until fairly recently pyqtgraph supported Qt4, which is probably why we haven't taken a close look at it.  If indeed the API is only accessible via QtQuick, it's highly unlikely that we will be incorporating it in the library in some way.
